### PR TITLE
feat(ci): add is_forked_pull_request parameter to CircleCI configuration

### PR DESCRIPTION
**Pull request by raffaele-oplabs**
This pull request introduces changes to the CircleCI configuration files to handle forked pull requests differently. The key changes include adding a new parameter to identify forked pull requests and modifying workflows to conditionally execute jobs based on this parameter.

Changes to CircleCI configuration:

* [`.circleci/artifact_upload_config.yml`](diffhunk://#diff-a0011e6b01ec3cc8297076866ff565f2a541b62752ad1f72f0ab03c85c29ca3fR13-R15): Added a new `is_forked_pull_request` parameter to the configuration.
* [`.circleci/config.yml`](diffhunk://#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L38-R43): Updated the job to generate configuration by setting the `is_forked_pull_request` parameter based on the branch name.
* [`.circleci/main_config.yml`](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619R13-R15): Added the `is_forked_pull_request` parameter and updated workflows to conditionally execute jobs based on this parameter. [[1]](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619R13-R15) [[2]](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619R413-R414) [[3]](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619R438-R451)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -181,9 +181,11 @@ See [Superchain Upgrades] OP Stack specifications.
...
-The following CircleCI checks are required and must pass before submitting a pull request:
+The following CircleCI checks are not mandatory for submitting a pull request, but they should be reviewed:
 
 - `ci/circleci: compute-genesis-diff`
 - `ci/circleci: compute-rollup-config-diff`
 
-If these checks did not run, they can be triggered manually from CircleCI without any parameters or by submitting and empty commit after the PR has being created.
+For pull requests from forks, these checks will not appear directly in the PR comments, but **the jobs will still run** and their results can be viewed in the diffs.
+
+Please note that while these jobs are **not blocking**, they must pass to ensure the accuracy of the changes. If the checks did not run, they can be triggered manually from CircleCI without any parameters or by submitting an empty commit after the PR has been created.
```
Comment (geoknee) [inline CONTRIBUTING.md:191]: Don't they run on every commit?

File: .circleci/main_config.yml [hunk 2]
```diff
@@ -430,3 +435,17 @@ workflows:
           message: "Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended"
           context: circleci-superchain-registry
           requires: [compute-rollup-config-diff]
+
+  compute-config-diff-and-report-forked-pr:
```
Comment (geoknee) [inline .circleci/main_config.yml:439]: Should we drop "report" from the name, because this never does any reporting?

**Review by geoknee — APPROVED** [review]
Thanks @raffaele-oplabs left a couple of comments.

File: CONTRIBUTING.md (hunk 1 above)
Comment (raffaele-oplabs) [reply CONTRIBUTING.md:191]: correct

File: .circleci/main_config.yml (hunk 2 above)
Comment (raffaele-oplabs) [reply .circleci/main_config.yml:439]: yes

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/50200).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/50199).

File: CONTRIBUTING.md (hunk 1 above)
Comment (geoknee) [reply CONTRIBUTING.md:192]: Then we don't need to say "If the checks did not run, they can be triggered manually from CircleCI without any parameters or by submitting an empty commit after the PR has been created." ?

**Pull request merged**